x/qgb/abci.go

```diff
@@ -113,3 +122,108 @@ func handleValsetRequest(ctx sdk.Context, k keeper.Keeper) {
 		}
 	}
 }
+
+func PruneIfNeeded(ctx sdk.Context, k keeper.Keeper) {
+	if !k.CheckLatestAttestationNonce(ctx) ||
+		k.CheckLatestAttestationNonce(ctx) && k.GetLatestAttestationNonce(ctx) <= AttestationPruningThreshold {
+		// If the attestations nonce hasn't been initialized yet, or the nonce is not greater than the minimum
+		// number of attestations, we don't need to prune anything
+		return
+	}
+
+	lastAvailableNonce := k.GetLastPrunedAttestationNonce(ctx) + 1
```

Comment:
I am open to suggestions, but I don't think `oldest` is any different since we will have to use it for other values as well to be consistent in naming

Reply:
> but I don't think oldest is any different since we will have to use it for other values as well to be consistent in naming

true, but it would help distinguish between last and latest, which is beneficial "last" is used to mean latest. such as "last time I went to the grocery store", I'm refering to the my latest visit

Reply:
Last and latest are used interchangeably in the code base, if I remember correctly. You have a point, we should have the same naming convention for all of them, and preferably decided by an English native person :D Personally, I didn't see a difference. 
What do you think we open an issue and change that? Changing the name wouldn't be state/consensus breaking, and only API breaking, so it's okey I guess and can be done after the feature freeze as part of the bug fixes and improvements

Reply:
https://github.com/celestiaorg/celestia-app/issues/1808